### ISSUE TITLE
Audio_transcribe hangs on long recordings

### ISSUE THREAD

**Issue by combinatorist** [Uberi/speech_recognition#130]
Steps to reproduce
------------------

1. I made adjustments to `examples/audio_transcribe.py` to work on a 6 second aif voice recording (I pulled out of the guts on a garage band "project" (`.band/Media/...aif`).
  * parameterized to file name argument from CLI
  * commented out everything but Sphinx recognizer
2. Then, I tried to run it on an hour long recording.

Expected behaviour
------------------

I expected it to run just fine on a longer file, but maybe it's not a linear algorithm or it doesn't buffer the results? 

Actual behaviour
----------------

 It took 9 hours, and 5 gigs of RAM before I gave up on it.  I can't share the actual file because it's confidential, but if you think I _should_ be able to transcribe hour long clips, I'll find another and reproduce the problem in more detail. Otherwise, any recommendations would be deeply appreciated (like what does an audio-phile call this sort of "streaming transcription"?).

System information
------------------

My **Python version** is 2.7.10.

My **SpeechRecognition library version** is 3.4.6.

My **PyAudio library version** is 0.2.9.

I **installed PocketSphinx from** Homebrew.

**Comment by combinatorist** [comment]
I've just added PR #133 to show the changes I made to `examples/audio_transcribe.py`

**Comment by Uberi** [comment]
That should definitely not be happening! If there's a non-confidental audio file you can reproduce the issue with, that would be very helpful.

As a temporary workaround, I recommend using `r.listen(source)` in a loop instead of `r.record(source)` to break the audio up into manageable sizes.

**Comment by combinatorist** [comment]
Ok, thanks for your feedback, Uberi!

I'll try to reproduce the issue with a redacted file.

**Comment by combinatorist** [comment]
Unfortunately, the file was too big for git, even after I switched to Git Large File Storage (or else I need some kind of special permissions for this repo):
```
Git LFS: (0 of 2 files) 0 B / 423.67 MB                                                                                                                        
@combinatorist can not upload new objects to public fork combinatorist/speech_recognition
Docs: https://github.com/contact
@combinatorist can not upload new objects to public fork combinatorist/speech_recognition
Docs: https://github.com/contact
error: failed to push some refs to 'https://github.com/combinatorist/speech_recognition.git'
```

I sent a support request to github, but meanwhile, I'm going to try to:
* create a public dropbox link to the big file
* create a shorter "long interview"
* switch the code to `listen` like you suggested

**Comment by combinatorist** [comment]
Ok, I updated the PR #133 with redacted files (however, you'll have to download the large one from my dropbox link).  Note: they sound like the real interviews, but they're pilot tests.

**Comment by combinatorist** [comment]
Thanks, @Uberi!  I experimented with `listen` and got it to work with my own `while` loop (see #133).  It's pretty hacky in it's current state [abd0523](https://github.com/Uberi/speech_recognition/pull/133/commits/abd0523498eb487047871911af2366fbddb07b7e), but `listen` **solves my original concerns,** so I'm closing this issue.  

Meanwhile, I'm happy to clean-up the version in my PR (#133) to make a helpful example for posterity.  My "short" interview (which fits on git), is enough to demonstrate the loop over listen (based on the timeout option). If you're interested, let's discuss it on the pull request.  I'm going to look for any develop docs now to see if you have any guidelines.

**Comment by RafahF** [comment]
Hello, I downloaded Python 2.7 and both PocketSphinx and SpeechRecognitions, but the code of Audio_Transcriber doesn't work .. 

I got this message: Sphinx error; missing PocketSphinx module: ensure that PocketSphinx is set up correctly.
Please note that I only changed the X.wav name and I wrote "__init__" instead of __file__.

Help me please.

**Comment by combinatorist** [comment]
Sorry, @RafahF, just saw your comment. 

I would be happy to help, but I need a couple things from you:
1. What code did you start with (link to a commit on github)?
2. What changes did you make (PR or diff or an explanation if the diff is really simple)?
3. Finally, if you're situation is different enough from mine, perhaps create a separate issue to document your environment etc.

Honestly, I don't know much about this `speech_recognition` library besides what I hacked around a while ago.  Furthermore, for my purposes, I switched from pocket sphinx to google speech API and never got around to including that into this library because I realized I could just use the google's example docs instead.

If you fill out the above information, I'll give it a shot this weekend when I have some down time.
